msal4j-sdk/src/test/java/com/microsoft/aad/msal4j/ManagedIdentityTests.java

```diff
@@ -149,7 +154,7 @@ void managedIdentityTest_UserAssigned_SuccessfulResponse(ManagedIdentitySourceTy
         IEnvironmentVariables environmentVariables = new EnvironmentVariablesHelper(source, endpoint);
         DefaultHttpClient httpClientMock = mock(DefaultHttpClient.class);
 
-        when(httpClientMock.send(eq(expectedRequest(source, resource, id)))).thenReturn(expectedResponse(200, getSuccessfulResponse(resource)));
+        lenient().when(httpClientMock.send(eq(expectedRequest(source, resource, id)))).thenReturn(expectedResponse(200, getSuccessfulResponse(resource)));
```

Comment:
Why do we use this lenient mocking instead of strict mocking? It feels like we're watering down these tests. 

If it's because of the static caching that MSI now has, which means that a token may be served from cache instead of from HTTP, then consider adding a test API to clear this cache. This is what .NET does and we can even discuss putting in the public API (maybe under some special namespace). 

Generally tests can follow the pattern: 

1. Clear internal cache
2. Setup HTTP mocks
3. Make first token acqusitions and ensure mocks are consumed
4. Make another token acquisition and ensure tokens come from cache.